src/libraries/Common/src/System/Net/NetworkInformation/HostInformationPal.Unix.cs

```diff
@@ -12,7 +9,12 @@ public static string GetHostName()
 
         public static string GetDomainName()
         {
-            return Interop.Sys.GetDomainName();
+            string domainName = Interop.Sys.GetDomainName();
+            if (domainName == "(none)")
+            {
+                return string.Empty;
+            }
+            return domainName;
```

Comment:
I think, this is not necessary, issue is reproducing on Linux, so Interop change should be enough.